NuGet.Packaging should reference System.Security.Cryptography.Pkcs v8 for net8.0 targets

### NuGet Product Used

NuGet SDK

### Product Version

NuGet.Packaging 7.0.0

### Worked before?

NuGet.Packaging 6.14.0

### Impact

I'm unable to use this version

### Repro Steps & Context

NuGet.Packaging 7.0.0 targets net8.0, but consumes [System.Security.Cryptography.Pkcs](https://www.nuget.org/packages/System.Security.Cryptography.Pkcs/) 9.0.6 instead of an 8.x era version.

This causes .NET 8 consumers to get MSB3277 warnings as I'll include in the verbose logs section.

While I _can_ workaround this by adding a direct package reference to `System.Formats.Asn1` 9.0.11, this means I have to ship one or more DLLs that should have come from the .NET runtime itself.

### Verbose Logs

```shell
Nerdbank.DotNetRepoTools.Tests net8.0 succeeded with 1 warning(s) (2.3s) → bin\Nerdbank.DotNetRepoTools.Tests\Debug\net8.0\Nerdbank.DotNetRepoTools.Tests.dll
    C:\Program Files\dotnet\sdk\10.0.100\Microsoft.Common.CurrentVersion.targets(2437,5): warning MSB3277:
      Found conflicts between different versions of "System.Formats.Asn1" that could not be resolved.
      There was a conflict between "System.Formats.Asn1, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" and "System.Formats.Asn1, Version=9.0.0.0, Culture=neutral, PublicKe
      yToken=cc7b13ffcd2ddd51".
          "System.Formats.Asn1, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" was chosen because it was primary and "System.Formats.Asn1, Version=9.0.0.0, Culture=neutral,
       PublicKeyToken=cc7b13ffcd2ddd51" was not.
          References which depend on "System.Formats.Asn1, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\8.0.22\re
      f\net8.0\System.Formats.Asn1.dll].
              C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\8.0.22\ref\net8.0\System.Formats.Asn1.dll
                Project file item includes which caused reference "C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\8.0.22\ref\net8.0\System.Formats.Asn1.dll".
                  C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\8.0.22\ref/net8.0/System.Formats.Asn1.dll
          References which depend on or have been unified to "System.Formats.Asn1, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [].
              Q:\src\DotNetRepoTools\bin\Nerdbank.DotNetRepoTools\Debug\net8.0\System.Security.Cryptography.Pkcs.dll
                Project file item includes which caused reference "Q:\src\DotNetRepoTools\bin\Nerdbank.DotNetRepoTools\Debug\net8.0\System.Security.Cryptography.Pkcs.dll".
                  Q:\src\DotNetRepoTools\bin\Nerdbank.DotNetRepoTools\Debug\net8.0\Nerdbank.DotNetRepoTools.dll
              Q:\src\DotNetRepoTools\bin\Nerdbank.DotNetRepoTools\Debug\net8.0\Microsoft.Bcl.Cryptography.dll
                Project file item includes which caused reference "Q:\src\DotNetRepoTools\bin\Nerdbank.DotNetRepoTools\Debug\net8.0\Microsoft.Bcl.Cryptography.dll".
                  Q:\src\DotNetRepoTools\bin\Nerdbank.DotNetRepoTools\Debug\net8.0\Nerdbank.DotNetRepoTools.dll
```

--- a/Directory.Packages.props
+++ b/Directory.Packages.props
@@ -4,7 +4,7 @@
 
     <MicrosoftCodeAnalysisPublicApiAnalyzersVersion Condition="'$(MicrosoftCodeAnalysisPublicApiAnalyzersVersion)' == ''">3.3.4</MicrosoftCodeAnalysisPublicApiAnalyzersVersion>
     <NewtonsoftJsonPackageVersion Condition="'$(NewtonsoftJsonPackageVersion)' == ''">13.0.3</NewtonsoftJsonPackageVersion>
-    <SystemFormatsAsn1PackageVersion Condition="'$(SystemFormatsAsn1PackageVersion)' == ''">9.0.6</SystemFormatsAsn1PackageVersion>
+    <SystemFormatsAsn1PackageVersion Condition="'$(SystemFormatsAsn1PackageVersion)' == ''">8.0.2</SystemFormatsAsn1PackageVersion>
 
     <!-- Read docs/updating-packages.md before updating System.Text.Json's version -->
     <SystemTextJsonVersion Condition="'$(SystemTextJsonVersion)' == ''">8.0.5</SystemTextJsonVersion>
@@ -27,8 +27,8 @@
     <MicrosoftWebXdtPackageVersion Condition="'$(MicrosoftWebXdtPackageVersion)' == ''">3.2.0</MicrosoftWebXdtPackageVersion>
     <SystemComponentModelCompositionPackageVersion Condition="'$(SystemComponentModelCompositionPackageVersion)' == ''">4.5.0</SystemComponentModelCompositionPackageVersion>
     <SystemMemoryPackageVersion Condition="'$(SystemMemoryPackageVersion)' == ''">4.6.3</SystemMemoryPackageVersion>
-    <SystemSecurityCryptographyPkcsVersion Condition="'$(SystemSecurityCryptographyPkcsVersion)' == ''">9.0.6</SystemSecurityCryptographyPkcsVersion>
-    <SystemSecurityCryptographyProtectedDataVersion Condition="'$(SystemSecurityCryptographyProtectedDataVersion)' == ''">9.0.6</SystemSecurityCryptographyProtectedDataVersion>
+    <SystemSecurityCryptographyPkcsVersion Condition="'$(SystemSecurityCryptographyPkcsVersion)' == ''">8.0.1</SystemSecurityCryptographyPkcsVersion>
+    <SystemSecurityCryptographyProtectedDataVersion Condition="'$(SystemSecurityCryptographyProtectedDataVersion)' == ''">8.0.0</SystemSecurityCryptographyProtectedDataVersion>
     <MicrosoftDotNetXliffTasksVersion Condition="'$(MicrosoftDotNetXliffTasksVersion)' == ''">10.0.0-beta.25260.104</MicrosoftDotNetXliffTasksVersion>
   </PropertyGroup>
 

